### PR TITLE
Deprecate color enum usage and change constants from ``str` to type `Color`

### PR DESCRIPTION
**Pull request by BrorSebastianSjovald**
## Overview: What does this pull request change?

This PR improves the color handling by deprecating uses of the Colors class and replace usage of the Enums with color constants as specified in https://github.com/ManimCommunity/manim/issues/2451 . At this point, a deprecation warning is raised when trying to use the Colors class using `manim.utils.deprecation`.

## Motivation and Explanation: Why and how do your changes improve the library?

The overall goal is to simplify the use of color by standardizing the approach as stated in https://github.com/ManimCommunity/manim/issues/2451 . By deprecating the use of the Enums and providing a way of using the constants in a similar way, we're creating a standard for how colors should be used.

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: manim/mobject/geometry/shape_matchers.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def construct(self):
     def __init__(
         self,
         mobject,
-        color: Colors | None = None,
+        color: str | None = None,
```
Comment (Darylgolden) [inline manim/mobject/geometry/shape_matchers.py:82]:
```suggestion
        color: Color | str | None = None,
```

Comment (Darylgolden) [reply manim/mobject/geometry/shape_matchers.py:82]: Pretty sure that a color object would work fine here too.

File: manim/mobject/mobject.py [hunk 2]
```diff
@@ -1646,7 +1639,7 @@ def put_start_and_end_on(self, start, end):
 
     # Background rectangle
     def add_background_rectangle(
-        self, color: Optional[Colors] = None, opacity: float = 0.75, **kwargs
+        self, color: str = None, opacity: float = 0.75, **kwargs
```
Comment (Darylgolden) [inline manim/mobject/mobject.py:1642]:
```suggestion
        self, color: Color | str | None = None, opacity: float = 0.75, **kwargs
```

**Review by Darylgolden — CHANGES_REQUESTED** [review]
Thank you for your work! Other than some minor typehint problems, I think this is good to go.

**Comment by behackl** [comment]
@BrorSebastianSjovald @ludwigjo any updates here?

**Comment by MrDiver** [comment]
Closing this for now. Making Colors the main source of truth again with exporting constants shouldn't break anything. `colour.py` has just too many duplicated features to our own implementation.

**Pull request closed without merging**